commons-geometry-euclidean/src/main/java/org/apache/commons/geometry/euclidean/twod/hull/ConvexHull2D.java

```diff
@@ -0,0 +1,469 @@
+     * heuristic to construct the hull. The heuristic is based on the idea of a
+     * convex quadrilateral, which is formed by four points with the lowest and
+     * highest x / y coordinates. Any point that lies inside this quadrilateral can
+     * not be part of the convex hull and can thus be safely discarded before
+     * generating the convex hull itself.
+     * <p>
+     * The complexity of the operation is O(n), and may greatly improve the time it
+     * takes to construct the convex hull afterwards, depending on the point
+     * distribution.
+     *
+     * @see <a href=
+     *      "http://en.wikipedia.org/wiki/Convex_hull_algorithms#Akl-Toussaint_heuristic">
+     *      Akl-Toussaint heuristic (Wikipedia)</a>
+     */
+    public static final class Builder {
+
+        /** Corner of triangle with minimal x coordinate. */
+        private Vector2D minX;
+
+        /** Corner of triangle with maximal x coordinate. */
+        private Vector2D maxX;
+
+        /** Corner of triangle with minimal y coordinate. */
+        private Vector2D minY;
+
+        /** Corner of triangle with maximal y coordinate. */
+        private Vector2D maxY;
+
+        /** Collection of all remaining candidates for a convex hull. */
+        private final Collection<Vector2D> candidates;
+
+        /** A precision context for comparing points. */
+        private final Precision.DoubleEquivalence precision;
+
+        /** Indicates if collinear points on the hull shall be present in the output.
+         * If {@code false}, only the extreme points are added to the hull.
+         */
+        private final boolean includeCollinearPoints;
+
+        /**Return a {@link Builder} instance configured with the given precision
+         * context. The precision context is used when comparing points.
+         *
+         * @param builderPrecision       precision context to use when building a convex
+         *                               hull from raw vertices; may be null if raw
+         *                               vertices are not used.
+         * @param includeCollinearPoints whether collinear points shall be added as hull
+         *                               vertices
+         */
+        public Builder(final boolean includeCollinearPoints, final Precision.DoubleEquivalence builderPrecision) {
+            this.precision = builderPrecision;
+            this.includeCollinearPoints = includeCollinearPoints;
+            candidates = EuclideanCollections.pointSet2D(builderPrecision);
+        }
+
+        /** Appends the given point to a collection of possible hull points, if and only
+         * if the given point is outside of a constructed quadrilateral of extreme properties.
+         *
+         * @param point a given point.
+         * @return this instance.
+         */
+        public Builder append(Vector2D point) {
+
+            //Checks if the given point supersedes one of the corners.
```

Comment:
Whitespace after `//`. Same again on the next comment below.

Reply:
Fixed